inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -94,33 +95,31 @@ private InstrumentationScope resolveScope(InstrumentationScopeSettings scopeSett
     }
 
     /**
-     * Creating a matcher for the given {@link NameMatcherSettings} which represents a superclass.
+     * Creating and adding a matcher for the given {@link NameMatcherSettings} which represents a superclass.
      */
-    private void processSuperclass(MatcherChainBuilder<TypeDescription> builder, NameMatcherSettings nameSettings) {
-        ElementMatcher.Junction<NamedElement> nameMatcher = nameIs(nameSettings);
-        if (nameMatcher != null) {
-            builder.and(hasSuperType(not(isInterface()).and(nameMatcher)));
+    private void processSuperclass(MatcherChainBuilder<TypeDescription> builder, DescriptionMatcherSettings descriptionSettings) {
+        ElementMatcher.Junction<TypeDescription> matcher = describedBy(descriptionSettings);
+        if (matcher != null) {
+            builder.and(hasSuperType(not(isInterface()).and(matcher)));
         }
     }
 
     /**
-     * Creating a matcher for the given {@link NameMatcherSettings} which represents an interface.
+     * Creating and adding a matcher for the given {@link NameMatcherSettings} which represents an interface.
      */
-    private void processInterface(MatcherChainBuilder<TypeDescription> builder, NameMatcherSettings nameSettings) {
-        ElementMatcher.Junction<NamedElement> nameMatcher = nameIs(nameSettings);
-        if (nameMatcher != null) {
-            builder.and(hasSuperType(isInterface().and(nameMatcher)));
+    private void processInterface(MatcherChainBuilder<TypeDescription> builder, DescriptionMatcherSettings descriptionSettings) {
+        ElementMatcher.Junction<TypeDescription> matcher = describedBy(descriptionSettings);
+        if (matcher != null) {
+            builder.and(hasSuperType(isInterface().and(matcher)));
         }
     }
 
     /**
-     * Creating a matcher for the given {@link NameMatcherSettings} which represents concrete classes.
+     * Creating and adding a matcher for the given {@link NameMatcherSettings} which represents concrete classes.
      */
-    private void processType(MatcherChainBuilder<TypeDescription> builder, NameMatcherSettings nameSettings) {
-        ElementMatcher.Junction<TypeDescription> nameMatcher = nameIs(nameSettings);
-        if (nameMatcher != null) {
-            builder.and(nameMatcher);
-        }
+    private void processType(MatcherChainBuilder<TypeDescription> builder, DescriptionMatcherSettings descriptionSettings) {
+        ElementMatcher.Junction<TypeDescription> nameMatcher = describedBy(descriptionSettings);
```

Comment:
Maybe also rename the variable "nameMatcher" here, as it does not only match by name anymore